vpr/src/pack/re_cluster_util.cpp

```diff
@@ -83,74 +107,61 @@ bool remove_atom_from_cluster(const AtomBlockId& atom_id,
             cleanup_pb(temp);
             has_more_children = count_children_pbs(next);
         }
+    }
 
-        //if the parents' names are the same as the removed atom names,
-        //update the name to prevent double the name when creating a new cluster for
-        // the removed atom
-        /*
-         * while(next != nullptr && *(next->name) == *atom_name) {
-         * next->name = vtr::strdup(child_name);
-         * if(next->parent_pb == nullptr)
-         * next = next->parent_pb;
-         * }
-         */
-
-        cluster_ctx.clb_nlist.block_pb(old_clb)->pb_route.clear();
-        cluster_ctx.clb_nlist.block_pb(old_clb)->pb_route = alloc_and_load_pb_route(router_data->saved_lb_nets, cluster_ctx.clb_nlist.block_pb(old_clb)->pb_graph_node);
-
-        if (during_packing) {
-            clustering_data.intra_lb_routing[old_clb] = router_data->saved_lb_nets;
-            router_data->saved_lb_nets = nullptr;
-        }
+    cluster_ctx.clb_nlist.block_pb(old_clb)->pb_route.clear();
+    cluster_ctx.clb_nlist.block_pb(old_clb)->pb_route = alloc_and_load_pb_route(old_router_data->saved_lb_nets, cluster_ctx.clb_nlist.block_pb(old_clb)->pb_graph_node);
 
-        else
-            get_imacro_from_iblk(&imacro, old_clb, g_vpr_ctx.placement().pl_macros);
-    } else {
-        VTR_LOG("re-cluster: Cluster is illegal after removing an atom\n");
+    if (during_packing) {
```

Comment:
Comment here would be a good idea.  Explain (as much as you understand) why this has to be updated during packing but not after an initial packing has been created.